### DISCUSSION TITLE
一点小小的建议，仅供作者参考

### DISCUSSION THREAD

**Discussion by caoxinyi1993** [MikeWang000000/Natter#59] [General]
工具很好用，已给star。谢谢作者写了这么简单好用的打洞工具。
总结了几个点，供作者今后更新时参考。
1，建议支持单程序多端口（范围）打洞
2，建议支持端口/ip存活自检测（如宽带重拨后自动检测变更ip重新打洞）
3，如果能联动server酱一类的通知程序，或者发邮件告知就更好了
4，写成服务，加个luci（x
可能多少有不靠谱的点，望见谅。

**Comment by MikeWang000000** [comment]
感谢反馈。目前来说 1～4 都是有希望达成的。

在 [v0.9](https://github.com/MikeWang000000/Natter/tree/v0.9) 分支中，需求 1～3 都是可以通过配置文件实现的。
不过 v0.9 还存在着bug，主要是 UDP 相关部分，因此还在开发中。
关于范围，Natter 可以通过配置文件将端口一个一个填上去，而不考虑大范围的打洞映射，因为这是一件很消耗资源的事情（需要维持连接），并且配太多的端口，失效的可能性高。

关于 luci 界面，已经有一位志愿者在工作了。
也怪我比较地摸鱼，更新并不是很频繁。目前生活中还有很多事需要处理。所以造成了现在实际上有很多其他人的修改版，也有 C 语言版的实现。XD

**Reply by despire1119** [reply]
大佬早上好，请问在2.0.0中如何实现需求1呢？现在我启动了多个服务，随机端口已被占用的概率变得有点高了。

**Comment by caoxinyi1993** [comment]
感谢回复。愿您线下诸事顺利。

**Comment by LordHumphrey** [comment]
> 

C实现版有地址不，没找到，谢谢啦~

**Comment by MikeWang000000** [comment]
> C实现版有地址不，没找到，谢谢啦~

在这呢 https://github.com/heiher/natmap
